inc/Engine/Media/PreloadFonts/AJAX/Controller.php

```diff
@@ -143,4 +146,36 @@ public function check_data(): array {
 
 		return $payload;
 	}
+
+	/**
+	 * Removes excluded fonts from the list of fonts to be preloaded.
+	 *
+	 * @param array $fonts Array of fonts to be preloaded.
+	 * @param array $exclusions Array of fonts to be excluded.
+	 *
+	 * @return array Filtered array of fonts, excluding those specified in the exclusion list.
+	 */
+	public function remove_excluded_fonts( array $fonts, array $exclusions ): array {
+		// Filter out the excluded fonts.
+		$filtered_fonts = array_filter(
+			$fonts,
+			function ( $font ) use ( $exclusions ) {
+				// Check if font is not in exclusions array.
+				if ( in_array( $font, $exclusions, true ) ) {
+					return false;
+				}
+
+				// Check if font doesn't contain any of the exclusion.
+				foreach ( $exclusions as $exclusion ) {
+					if ( stripos( $font, $exclusion ) !== false ) {
+						return false;
+					}
+				}
+
+				return true;
+			}
+		);
+
+		return array_values( $filtered_fonts );
```

Comment:
The time complexity is approximately O(n × m), where:
- n = number of $fonts
- m = number of $exclusions

I think we need to move this to be regex so we will combine all regex elements into one regex (imploded by `|`) and compare it with each font in the loop, something like:

```suggestion
	// Escape special regex characters in exclusions and implode them into a regex pattern
    $escaped_exclusions = array_map(function ($exclusion) {
        return preg_quote($exclusion, '/');
    }, $exclusions);

    // Combine all exclusions into one regex pattern, separated by `|`
    $regex_pattern = '/(' . implode('|', $escaped_exclusions) . ')/i';

    // Filter fonts by checking against the regex pattern
    $filtered_fonts = array_filter(
        $fonts,
        function ($font) use ($regex_pattern) {
            return !preg_match($regex_pattern, $font);
        }
    );

    // Reset array keys and return the filtered fonts
    return array_values($filtered_fonts);
```

This will change the time complexity to be O(n), where n is the number of $fonts.
What do u think?